### DISCUSSION TITLE
Creating an org-capture template items (i.e. tasks)  in my denote inbox-file

### DISCUSSION THREAD

**Discussion by ghost** [protesilaos/denote#431] [Q&A]
In my workflow I write down tasks and have them stored as TODO items in an inbox-file. 

I used to do it with an org-roam-capture template:

```
("i" "inbox" plain
"* TODO %^{Title|${title}} %^G\n%^{time|SCHEDULED:|DEADLINE:} %^t\n:PROPERTIES:\n:CAPTURED: %U\n:END:\n%?"
 :target (file+head "Inbox.org"
                  "#+TITLE: Inbox\n")
  :empty-lines 1
 )
```

I was able to re-create the template using a function which is now storing the TODO items a denote file:

```
(defun lcb/create-todo-entry ()
  "Create a TODO entry in a specified inbox file."
  (interactive)
  (let* ((todo-file (concat denote-directory "20240911T222826==meta--inbox__metanote.org")) ; Change this to your TODO file path
         (title (read-string "Enter TODO title: "))
         (schedule-or-deadline (completing-read "SCHEDULED or DEADLINE? " '("SCHEDULED:" "DEADLINE:")))
         (date (org-read-date nil nil nil "Select date: ")) ; Opens calendar for date selection
         (additional-info (read-string "Enter additional information: ")) ; New prompt for additional info
         (entry-text (format "* TODO %s\n  %s <%s>\n %s\n"
                             title
                             schedule-or-deadline
                             date
                             additional-info)))

    ;; Append the new entry to the specified TODO file
    (with-temp-buffer
      ;; Insert the new entry at the end of the buffer
      (insert entry-text)
      ;; Write it back to the todo file, appending it
      (write-region (point-min) (point-max) todo-file t))
    ;; Revert buffer to reflect changes if it's open in another window or buffer.
    (when-let ((buf (find-buffer-visiting todo-file)))
      (with-current-buffer buf
        (revert-buffer t t)))))

```
But since I don’t like custom functions like this and would favor org-capture template for it, 
I was wondering how I would write a proper denote-org-capture template for this. 

------ EDIT

I was giving it a try with `denote-templates`:

```
(setq denote-templates
      `((inbox . lcb/create-todo-entry)
        ))
```
but it creates first a file and prompting for a title etc and then it calls the function.

**Comment by protesilaos** [comment]
> From: "Lukas C. Bossert" ***@***.***>
> Date: Wed, 11 Sep 2024 14:32:16 -0700
>
> In my workflow I write down tasks and have them stored as TODO items in an inbox-file. 
>
> I used to do it with an org-roam-capture template:
>
> ```
> ("i" "inbox" plain
> "* TODO %^{Title|${title}} %^G\n%^{time|SCHEDULED:|DEADLINE:} %^t\n:PROPERTIES:\n:CAPTURED: %U\n:END:\n%?"
>  :target (file+head "Inbox.org"
>                   "#+TITLE: Inbox\n")
>   :empty-lines 1
>  )
> ```

> [... 32 lines elided]

If I understand this correctly, you are creating a new entry inside of
the same file, not a new file each time, right? If so, why not do this
with the regular org-capture-templates?

Denote does not need to be involved in this step. Then, if you need to
split out the heading into its own standalone file, you can use the
command 'denote-org-extras-extract-org-subtree'.

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by ghost** [comment]
You are quite right - it doesnt make a lot of sense to write a function and make things more complicated. I set the new file as destination which has the denote naming-scheme. Everything is fine now.
